Standardize component config introductions/options

conceptualshark opened the pull request:
## Description

An attempt to start the component configuration pages on the path of some sort of standardization. This was scoped down to only tackle the introduction and config options for each page, and is not meant to tackle bigger areas of restructuring, rewriting, etc.

## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [x] There is **no urgency** with this change and can be released at any time.

## PR Checklist

<!-- Camunda maintains 18 months of versions. Backporting your change to multiple versions is common. -->

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [ ] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Adding or removing pages requires extra steps.
- [ ] I included my new page in the sidebar file(s).
- [ ] I added a redirect for a renamed or deleted page to the .htaccess file.
-->

<!-- All changes require either an Engineering review or technical writer review. **Many require both!** -->

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Changes to **docs infra**, including updates to workflows and adding new npm packages, must be first discussed via issue or #ask-c8-documentation and linked for context.
- [ ] My changes require a [docs infrastructure review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process). -->

christinaausley commented:
CC @camunda/distribution for review here 👍

akeller commented on docs/self-managed/console-deployment/configuration.md:
```diff
@@ -9,14 +9,21 @@ description: "Read details on the configuration variables of Console Self-Manage
 :::
 
-Console Self-Managed can be configured using environment variables and configuration parameters.
+Console can be configured using environment variables, configuration parameters, or a combination of both. When configuring your Console setup, keep in mind the following:
```
This is probably a PM question, but do we want to emphasize "Console Self-Managed" as the component name? Or are they similar enough that we can drop "Self-Managed" with the right context?

hamza-m-masood replied:
I think it is a good idea to specify self-managed as well, since it is a different development stream. It is quite different from SaaS.
cc @theburi

hamza-m-masood commented on docs/self-managed/console-deployment/configuration.md:
````diff
@@ -71,7 +80,7 @@ console:
 ```
 
-## Using a different OpenID Connect (OIDC) authentication provider than Keycloak
+## Configure an OpenID Connect (OIDC) provider
````
I think it is still a good idea to mention keycloak somewhere in the title, just so it is clear that the OIDC provider and keycloak are mutually exclusive. 
Or maybe mention it in a paragraph under the title. 
What do you think @conceptualshark ?

hamza-m-masood commented on docs/self-managed/identity/deployment/configuration-variables.md:
```diff
@@ -5,10 +5,18 @@ sidebar_label: "Configuration variables"
+## Environment variables
+
+:::note
+Underscores in environment variables correspond to configuration file key levels.
```
Maybe this is true for all configuration options in identity, but I don't think it holds true for other components. I think it would be better to use `Generally,...` at the start

akeller commented:
Removing `deploy` since it's currently broken. 😢

conceptualshark commented:
Removed the less-certain pieces of this update (if underscores always represent key levels) and included feedback re: Keycloak specificity, specifying Console Self Managed. Also removed a reference to Console being enterprise-specific and will follow up on other instances of that in a separate issue.

christinaausley reviewed (approved):
I love this! 🥳

akeller reviewed (commented):
👍 No concerns.

I did notice some repeated content that could be added to a partial/reusable component, but would consider that a good item to follow up on.

hamza-m-masood reviewed (changes requested):
I have raised my concerns in this comment:
https://github.com/camunda/camunda-docs/pull/4189#discussion_r1877974620
